src/ip_range_input.rs

```diff
@@ -0,0 +1,165 @@
+use crate::{input::view_asterisk, language::Language, text, IpRange, Texts};
+use json_gettext::get_text;
+use std::{cell::RefCell, net::Ipv4Addr, rc::Rc, str::FromStr};
+use wasm_bindgen::JsCast;
+use web_sys::HtmlInputElement;
+use yew::{html, Component, Context, Html, InputEvent, Properties};
+
+const DEFAULT_MAX_HEIGHT: u32 = 280;
+
+#[derive(Clone, PartialEq, Eq)]
+pub enum Message {
+    InputString(String),
+    InputError,
+}
+
+pub struct Model {
+    input: String,
+}
+
+#[derive(Clone, Eq, PartialEq, Properties)]
+pub struct Props {
+    pub txt: Texts,
+    pub language: Language,
+    #[prop_or(None)]
+    pub width: Option<u32>,
+    #[prop_or(DEFAULT_MAX_HEIGHT)]
+    pub max_height: u32,
+    pub input_data: Rc<RefCell<Option<IpRange>>>,
+    #[prop_or(None)]
+    pub title: Option<String>,
+    #[prop_or(None)]
+    pub example_message: Option<String>,
+    #[prop_or(None)]
+    pub placeholder_message: Option<String>,
+    #[prop_or(false)]
+    pub required: bool,
+}
+
+impl Component for Model {
+    type Message = Message;
+    type Properties = Props;
+
+    fn create(_ctx: &Context<Self>) -> Self {
+        Self {
+            input: String::new(),
+        }
+    }
+
+    fn changed(&mut self, _ctx: &Context<Self>, _old_props: &Self::Properties) -> bool {
+        true
+    }
+
+    fn update(&mut self, ctx: &Context<Self>, msg: Self::Message) -> bool {
+        match msg {
+            Message::InputString(text) => {
+                self.input = text.clone();
+                if let Ok(mut data) = ctx.props().input_data.try_borrow_mut() {
+                    if let Some(range) = check_input(&text) {
+                        data.replace(range);
+                    }
+                }
```

Comment:
If the line starting with `self.input` can be after the `if let`, it doesn't need to use `clone`.

Reply:
Regardless of `clone`, I am wondering what would happen if you enter any wrong IP range and click search. `ctx.props().input_data` still has a valid range if it was given any previous one and search will be being processed even though there is a message showing wrong input. I think you need to find a way to prevent it happening.

Reply:
First comment done. I've made it so that if there was incorrect input, I return None to ctx.props().input_data and on aice-web side it should have None for the ranges.

Reply:
You mean, even if search is clicked, no further process with None?

Reply:
It would have the same effect as if the filter on the nice-web side had orig_addr: None, orig_port: None, or the other ranges as None. So same effect if there are no inputs when searched is clicked. I'll present some images to demonstrate. I think that if I handle None to not process, it might also interfere if the user does not add anything into range.

Reply:
Initial Load:
<img width="1512" alt="Screenshot 2023-05-16 at 4 25 22 PM" src="https://github.com/aicers/frontary/assets/102562226/9606f94d-205a-418f-9308-466a2b86534b">

With Filter and correct input:
<img width="1512" alt="Screenshot 2023-05-16 at 4 25 32 PM" src="https://github.com/aicers/frontary/assets/102562226/9c2b4e99-2655-4b62-baeb-67028db0807d">

With Filter and incorrect input:
<img width="1512" alt="Screenshot 2023-05-16 at 4 25 42 PM" src="https://github.com/aicers/frontary/assets/102562226/910b4284-0155-440f-a1f3-3ce790715da9">

With incorrect input (same as initial load):
<img width="1512" alt="Screenshot 2023-05-16 at 4 25 49 PM" src="https://github.com/aicers/frontary/assets/102562226/5497e6c8-05a1-4297-bdff-5131d01234e0">

Reply:
So with the 1st and 4th images they are the same since the Details section part of the filter (all 4 inputs)  is None